plugins/palo_alto_cortex_xdr/icon_palo_alto_cortex_xdr/tasks/monitor_alerts/task.py

```diff
@@ -0,0 +1,381 @@
+import insightconnect_plugin_runtime
+from .schema import (
+    MonitorAlertsInput,
+    MonitorAlertsOutput,
+    MonitorAlertsState,
+    Input,
+    Output,
+    Component,
+    State,
+)
+import time
+from datetime import datetime, timedelta, timezone
+from insightconnect_plugin_runtime.exceptions import PluginException, APIException
+from insightconnect_plugin_runtime.helper import response_handler, extract_json, hash_sha1, make_request
+from typing import Any, Dict, Tuple
+import requests
+import urllib
+
+
+TIME_FORMAT = "%Y-%m-%dT%H:%M:%S.%fZ"
+MAX_LOOKBACK_DAYS = 7
+DEFAULT_LOOKBACK_HOURS = 24
+ALERT_LIMIT = 100
+
+MAX_LIMIT = 7500
+
+# State held values
+LAST_ALERT_TIME = "last_incident_time"
+LAST_ALERT_HASH = "last_incident_hash"
+# State held values = CONOR
+LAST_SEARCH_FROM = "last_search_from"
+LAST_SEARCH_TO = "last_search_to"
+
+# Used to paginate
+ALERTS_OFFSET = "alerts_offset"
+FROM_TIME_FILTER = "from_time_filter"
+TO_TIME_FILTER = "to_time_filter"
+TIMESTAMP_KEY = "event_timestamp"
+
+NEXT_PAGE_LINK = "next_page_link"
+# Custom imports below
+
+
+class MonitorAlerts(insightconnect_plugin_runtime.Task):
+    def __init__(self):
+        super(self.__class__, self).__init__(
+            name="monitor_alerts",
+            description=Component.DESCRIPTION,
+            input=MonitorAlertsInput(),
+            output=MonitorAlertsOutput(),
+            state=MonitorAlertsState(),
+        )
+
+    def run(self, params={}, state={}, custom_config: dict = {}):  # pylint: disable=unused-argument
+        existing_state = state.copy()
+        has_more_pages = False
+        parameters = {}
+        print(f"{existing_state = }")
+
+        custom_config = {}
+        state = {LAST_SEARCH_FROM: None, LAST_SEARCH_TO: None}
+
+        try:
+
+            now_time = self._get_current_time()
+            now = now_time - timedelta(minutes=15)
+            # last 15 minutes
+            end_time = now.strftime(TIME_FORMAT)
+            print(f"{end_time = }")
+            start_time, alert_limit = self._parse_custom_config(custom_config, now, state)
+
+            self.logger.info("Starting to download alerts...")
+
+            response, state, has_more_pages = self.get_alerts_palo_alto(state=state, custom_config=custom_config)
+
+            # TODO - Are we supposed to show the total (MAX LIMIT) or the total (total_count)
+            self.logger.info(f"Total alerts returned: {len(response)}")
+
+            # Local Debugging
+            self.logger.info(f"{type(response) = }")
```

Comment:
Some leftover log statements.

Reply:
Debugging statements while we work on it